### ISSUE TITLE
On both 2.7.12 and 3.6.0a3, affirm seems to do nothing.

### ISSUE THREAD

**Issue by pobocks** [elifiner/affirm#1]
It successfully installs via pip, but there's no change to output of assertions, whether through assert() or affirm().

I'm on OSX 10.9.5, Pythons provided via brew, system, and virtualenv all seem to have same issue.

**Comment by finereli** [comment]
Interesting. Can you share a sample that shows this? I can't reproduce it here.
Can you also clone the repo and run `test.sh` on your system to see if the tests pass?

**Comment by pobocks** [comment]
My test case was basically piping the examples described in the docs to python -c, and running them in default console and Jupyter.

Cloning and running test now!

**Comment by pobocks** [comment]
Tests pass AFAICT.

So, I actually get different results running from file... 
1. Doing a whole module import, I get "None" as the message value for the assertion rather than variable values
2. It works with `from affirm import affirm`

**Comment by finereli** [comment]
Yeah, neither affirm nor assert work without the code being stored in a file. To figure out the original expression, I load the source file and look for it there. So running with `python -c` or in interactive mode doesn't work.

`import affirm` should work for regular assert statements, let me know if it doesn't work for you.

Also, really nice to hear somebody is using my tiny library, thanks!

**Comment by pobocks** [comment]
Ah, that makes sense of why case 2 works now, but 'import affirm + regular
assert' is still reporting None instead of the generated message. I'll poke
around some.

Hmmmm. I wonder if it would be possible to fall back to inspecting function
args or locals or such?  I have up till now done more ruby, so I don't know
the details on inspecting bindings at place of execution.

You're very welcome! I'll let you know if I can figure out anything else
here.

**Comment by finereli** [comment]
It's much harder to do it without source being available since you need to
parse bytecode. I didn't want to have affirm depend on something like
python-meta which can do that but is quite an overkill.

I previously started working on a bytecode parser, but didn't get far
enough for it to be useful. I'll take another look, maybe some of it can be
integrated into affirm.
